run.ps1

```diff
@@ -0,0 +1,30 @@
+# set the base tools directory
+$toolsLocalPath = Join-Path $PSScriptRoot ".tools"
+$bootStrapperPath = Join-Path $toolsLocalPath "bootstrap.ps1"
+
+# if the boot-strapper script doesn't exist then download it
+if ((Test-Path $bootStrapperPath) -eq 0)
+{
+    if ((Test-Path $toolsLocalPath) -eq 0)
+    {
+        mkdir $toolsLocalPath | Out-Null
+    }
+
+    # download boot-strapper script
+    Invoke-WebRequest "https://raw.githubusercontent.com/dotnet/buildtools/master/bootstrap/bootstrap.ps1" -OutFile $bootStrapperPath
+}
+
+# now execute it
+Invoke-Expression "$bootStrapperPath (Get-Location) $toolsLocalPath" | Out-File (Join-Path (Get-Location) "bootstrap.log")
+if ($LastExitCode -ne 0)
+{
+    Write-Output "Boot-strapping failed with exit code $LastExitCode"
+    exit $LastExitCode
+}
+
+# execute the tool using the dotnet.exe host
+$dotNetExe = Join-Path $toolsLocalPath "dotnetcli\shared\Microsoft.NETCore.App\version\dotnet.exe"
+$runExe = Join-Path $toolsLocalPath "Microsoft.DotNet.BuildTools.Run\netcoreapp1.0\run.exe"
+$process = Start-Process -NoNewWindow -FilePath $dotNetExe -ArgumentList "$runExe $args" -PassThru
```

Comment:
I would still like to avoid using Start-Process and instead just do `& $dotNetExe $runExe $args` then we don't need to worry about what the different semantics of `-Wait` vs `WaitForExit`. I don't think there is any reason we need to get a process object here. We should be able to just call the command and then call `exit $LastExitCode`.